Azure/sonic-buildimage
# DellEMC S6100 BIOS upgrade support

**- What I did**
- Added BIOS upgrade support for DellEMC S6100 platform.

**- How I did it**
- flahrom utitily is needed for BIOS upgrade support in S6100. 
- flashrom and associated patch file is added in vendor section. "flashrom" binary will be generated at the time of build-creation. 
- Dell specific BIOS image is uploaded in appropriate device directory.
- To upgrade BIOS below command needs to be executed:
   ```flashrom -p internal -w /usr/share/sonic/device/x86_64-dell_s6100_c2538-r0/bin/S6100-BIOS-3.25.0.2-7.bin```

**- How to verify it**
- You can check the BIOS version in boot logs:
```
BIOS Boot Selector for S6100
Primary BIOS Version 3.25.0.2-7<----version
SMF Version: MSS 2.1, FPGA 1.1
Last POR=0x11, Reset Cause=0x44


POST Configuration
  CPU Signature 406D8
```
- or you can use below command
```
root@sonic:/home/admin# dmidecode -s system-version
3.25.0.2-7
```
